.github/workflows/tests.yml

```diff
@@ -0,0 +1,29 @@
+name: Run tests
+
+# Trigger the workflow on push or pull request
+on: [push, pull_request]
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+      # Set everything up
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+      - run: git submodule update --init --recursive
+
+      # Run tests and whatnot
+      - run: npm install
+      - run: npm run build
+      - run: npm run lint
+      - run: npm run coverage
+        env:
+          CONTINUOUS_INTEGRATION: 1
+
+      # Upload coverage data
+      - name: Coveralls
+        if: ${{ success() || failure() }}
```

Comment:
this shouldn't upload if this is a push to master and the tests fail

Reply:
This only uploads coverage data, which should still be uploaded even with failing tests.